lib/wiki_assignment_output.rb

```diff
@@ -56,11 +54,10 @@ def build_talk_page_update
   ###################
   # Helper methods #
   ###################
-  # This is to avoid creating a new empty talk page.
-  def prevent_removal_of_assignments_from_non_existent_talk_pages(talk_page_does_not_exist)
-    # If assignments are empty, that means to remove assignments
-    return unless @assignments.empty? && talk_page_does_not_exist
-    yield
+  # Normal article means the article page actually exists, and is not a disambiguation page.
+  def normal_article?
+    article_content = WikiApi.new(@wiki).get_page_content(@title)
+    article_content.present? && !includes_disambiguation_template?(article_content)
```

Comment:
Another option would be to use the negation of `or` operator, but I feel like `and` operator is usually easier to understand:
`!(article_content.blank? || includes_disambiguation_template?(article_content))`

Reply:
Your implementation looks good to me.